applications/asset_tracker_v2/src/modules/data_module.c

```diff
@@ -1369,73 +1359,54 @@ static void on_all_states(struct data_msg_data *msg)
 		return;
 	}
 
-	if (IS_EVENT(msg, gnss, GNSS_EVT_DATA_READY)) {
-		struct cloud_data_gnss new_gnss_data = {
-			.gnss_ts = msg->module.gnss.data.gnss.timestamp,
-			.queued = true,
-			.format = msg->module.gnss.data.gnss.format
-		};
-
-		switch (msg->module.gnss.data.gnss.format) {
-		case GNSS_MODULE_DATA_FORMAT_PVT: {
-			/* Add PVT data */
-			new_gnss_data.pvt.acc = msg->module.gnss.data.gnss.pvt.accuracy;
-			new_gnss_data.pvt.alt = msg->module.gnss.data.gnss.pvt.altitude;
-			new_gnss_data.pvt.hdg = msg->module.gnss.data.gnss.pvt.heading;
-			new_gnss_data.pvt.lat = msg->module.gnss.data.gnss.pvt.latitude;
-			new_gnss_data.pvt.longi = msg->module.gnss.data.gnss.pvt.longitude;
-			new_gnss_data.pvt.spd = msg->module.gnss.data.gnss.pvt.speed;
-
+	if (IS_EVENT(msg, location, LOCATION_MODULE_EVT_DATA_READY)) {
+		struct cloud_data_gnss new_location_data = {
+			.gnss_ts = msg->module.location.data.location.timestamp,
```

Comment:
Note to myself: Need to check what is this gnss_ts time in cellular case

Reply:
We only hit this `LOCATION_MODULE_EVT_DATA_READY` branch with GNSS currently as cellular goes to `LOCATION_MODULE_EVT_NEIGHBOR_CELLS_DATA_READY` branch few lines below.
If cellular will hit this, we need to check the timestamp.